### PR TITLE
Token registry

### PR DESCRIPTION
**Pull request by anondev2323**
- Script checks for invalid values in config and throw error if found
- Script compiles a list of updates
- `checkForeignAssetConfig` checks for updates and logs the changes needed
- `updateForeignAssetConfig` updates the tokens list
- Add CI for determining if the list is out of date

TODO (gonna handle this [later](https://github.com/wormhole-foundation/connect-sdk/issues/91)):
- [ ] Check this list before fetching foreign assets/decimals

### CODE REVIEW

File: core/tokenRegistry/README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+- "NftBridgeRecoverChainId"
+- "RelayerRegisterChain"
+- "RelayerUpgradeContract"
+- "RelayerUpdateDefaultProvider"
+
+Circle
+- "CircleTransferRelay"
+
+BAM
+- "BAMessage"
```
Comment (barnjamin) [inline core/tokenRegistry/README.md:52]: guessing this file shouldn't be copied over?

File: core/tokenRegistry/src/foreignAssets.ts [hunk 2]
```diff
@@ -0,0 +1,132 @@
+// patch out annoying logs
+const info = console.info;
+console.info = function (x: any, ...rest: any) {
+  if (x !== 'secp256k1 unavailable, reverting to browser version') {
+    info(x, ...rest);
+  }
+};
```
Comment (barnjamin) [inline core/tokenRegistry/src/foreignAssets.ts:7]: incredible 🥇

File: core/tokenRegistry/src/foreignAssets.ts [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+  ) {
+    warn(x, ...rest);
+  }
+};
+
+import { ChainName, chains } from "@wormhole-foundation/sdk-base";
+import { ForeignAssetsCache, TokensConfig } from "./types";
+import { TokenId, Wormhole, toNative } from "@wormhole-foundation/connect-sdk";
+
+// TODO: Question: How do we handle if a user tries to perform an action for a chain/platform which isn't installed??
```
Comment (barnjamin) [inline core/tokenRegistry/src/foreignAssets.ts:25]: I vote puke hard w/ msg about installing that package

File: core/tokenRegistry/src/scripts/checkForeignAssetConfig.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
+// patch out annoying logs
+const info = console.info;
+console.info = function (x: any, ...rest: any) {
+  if (x !== 'secp256k1 unavailable, reverting to browser version') {
+    info(x, ...rest);
+  }
+};
```
Comment (barnjamin) [inline core/tokenRegistry/src/scripts/checkForeignAssetConfig.ts:7]: do we have to do this in every file we want it patched for?

File: core/tokenRegistry/src/foreignAssets.ts (hunk 3 above)
Comment (anondev2323) [reply core/tokenRegistry/src/foreignAssets.ts:25]: Def agree, broke it out into this issue https://github.com/wormhole-foundation/connect-sdk/issues/90

File: core/tokenRegistry/src/scripts/checkForeignAssetConfig.ts (hunk 4 above)
Comment (anondev2323) [reply core/tokenRegistry/src/scripts/checkForeignAssetConfig.ts:7]: Not sure 🤔

File: .github/workflows/checkForeignAssetsConfig.yml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 20
+          cache: "npm"
+      - run: npm ci
+      - run: npm run build --if-present
+      - run: cd core/tokenRegistry && npx ts-node src/scripts/checkForeignAssetsConfig.ts
```
Comment (barnjamin) [inline .github/workflows/checkForeignAssetsConfig.yml:24]: could/should we add this to the existing one so we don't have to wait for it to rebuild another env?

File: connect/src/wormhole.ts [hunk 6]
```diff
@@ -259,7 +259,7 @@ export class Wormhole {
    */
   async getDecimals(
     chain: ChainName,
-    token: TokenId | "native",
+    token: NativeAddress<any> | UniversalAddress | "native",
```
Comment (barnjamin) [inline connect/src/wormhole.ts:262]: any here because of some issue w/ using `PlatformName`?

Comment (anondev2323) [reply connect/src/wormhole.ts]: Just lazy really 😅 Updated it

**Review by barnjamin — APPROVED** [review]

**Pull request merged**